### ISSUE TITLE
Compile guide?

### ISSUE THREAD

**Issue by mdhari** [jmacd/xdelta#213]
Sorry if I haven't looked hard enough. I'm trying to figure out how to compile this on my Mac OSX with El Capitan. Usually I'm able to run these commands:

`./configure`
`make install`

But it's failing on the first command:

`./configure.ac` 
`./configure.ac: line 1: syntax error near unexpected token '[Xdelta3],'`
`./configure.ac: line 1: 'AC_INIT([Xdelta3], [3.1.0], [josh.macdonald@gmail.com], '`

Are there any helpful links that you could point me to for compiling this?

**Comment by fenugrec** [comment]
Have you tried running "automake --add-missing" and possibly autoreconf  before "./configure" ?

**Comment by mdhari** [comment]
Is autoremake another program I need to download?

**Comment by fenugrec** [comment]
Sorry I typo'd, it's automake and autoreconf. Part of GNU autotools

**Comment by lgpmichael2** [comment]
If it helps, this is what I've managed to put together to build xdelta on Centos 6, which isn't using most of the stuff that xdelta3 requires (for some reason, not sure why xdelta which simply handles bytes needs the latest of anything, but there you go).

Depending on your problems, some or more of these steps may not be needed. However, it does give a pretty good idea of the steps needed where almost nothing is of the version needed, and nothing works.

This uses the sources from the gnu website.

If you get errors about Makefile.in mission, install libtool

```
#!/bin/sh -                                                                                 

rm -rf autoconf-2.69
rm -rf autoconf-bin

tar -xvf autoconf-2.69.tar.gz

cd autoconf-2.69

./configure --prefix=`pwd`/../autoconf-bin
make
make install

rm -rf automake-1.15
rm -rf automake-bin

cd ..

tar -xvf automake-1.15.tar.gz

cd automake-1.15

PATH=../autoconf-bin/bin:$PATH ./configure --prefix=`pwd`/../automake-bin AUTOCONF=`pwd`/..\
/autoconf-bin/bin/autoconf
make
make install

cd ..

rm -rf xdelta-3.1.0

tar xvf xdelta-3.1.0.tar.gz

cd xdelta-3.1.0/xdelta3

PATH=../../autoconf-bin/bin/:$PATH ../../automake-bin/bin/aclocal
../../autoconf-bin/bin/autoconf
../../autoconf-bin/bin/autoheader

PATH=../../autoconf-bin/bin/:$PATH ../../automake-bin/bin/automake --add-missing

./configure

cat Makefile | sed -e "s/11/0x/g" > Makefile.new
mv -f Makefile.new Makefile


make
```

**Comment by watercrossing** [comment]
These instructions didn't work for me.  I had to set up my own set of m4, autoconf, automake and  libtool; and then run 

```
libtoolize
aclocal
autoconf
automake --add-missing
./configure --prefix=$HOME/tools
```

The call to `libtoolize` seems to be missing from most instructions I have found on the web.

**Comment by samuelcolvin** [comment]
Thanks so much @watercrossing. After numerous failed attempt that worked for me on Ubuntu 16.04.

**Comment by samuelcolvin** [comment]
Actually, turns out `autoheader` before `automake --add-missing` is required too.

**Comment by zmsoft** [comment]
@mdhari Did you compile successfully under Mac?What is the compile step?

**Comment by mdhari** [comment]
Nah, I don't think so. I think I tried an older compiled version on the net.

**Comment by zmsoft** [comment]
@mdhari Can you share it?Many thanks

**Comment by mdhari** [comment]
It looks like it's on homebrew

http://brewformulas.org/xdelta

Try brew install xdelta

**Comment by codemerx** [comment]
Here is how to build it - https://github.com/jmacd/xdelta/issues/235#issuecomment-468763347

**Comment by jmacd** [comment]
Addressed by #283: the project now builds with CMake and the README documents the build steps (xdelta3/README.md). Closing as resolved; re-open if anything is unclear. Thanks!
